src/Expensify.js

```diff
@@ -134,6 +138,9 @@ class Expensify extends PureComponent {
 
             // eslint-disable-next-line react/no-did-update-set-state
             this.setState({isSplashShown: false});
+
+            // If the app is opened from a deep link, get the reportID (if exists) from the deep link and navigate to the chat report
+            Linking.getInitialURL().then(url => ReportUtils.openReportFromDeepLink(url));
```

Comment:
Ah we call this method here to open the chat when the user opens a deep link, and the app is opened (that's why it's in the `componentDidUpdate` method). We also have this method in the `componentDidMount` already when the app is closed, and the user clicks on a deep link.

Also we don't [open the report immediately](https://github.com/Expensify/App/blob/ef547075af5f3fb33508aac453b248d5e6e8708c/src/libs/actions/Report.js#L1386-L1399), we wait until the navigation is complete. 

If we update this logic, we'd need to ensure the test steps listed [here](https://github.com/Expensify/App/pull/14669) are still passing.

Reply:
> Do you mean when app is fully loaded?

Yup

> Linking.getInitialURL is usually used when the app is just mounted (not open before), and Linking.addEventListener is used when app is opened and running in background

I recall this was added for an edge case in mobile browsers or redirecting the user to the chat after they open the deep link and sign in. But happy that we can implement it in the correct way, but just we'd need to ensure it still redirects the user to the chat on all platforms (if it's already authenticated or after authentication), otherwise it could cause a regression.